common/src/main/java/com/wynntils/wynn/model/bossbar/BossBarModel.java

```diff
@@ -0,0 +1,219 @@
+                public void setProgress(float progress) {
+                    super.setProgress(progress);
+
+                    if (targetProgress != 0f) {
+                        // Round to nearest 30
+                        int unroundedMax = (int) (current / targetProgress);
+                        int remainder = unroundedMax % 30;
+
+                        max = unroundedMax - remainder;
+                        if (remainder > 15) {
+                            max += 30;
+                        }
+                    }
+                }
+            };
+
+    public static final TrackedBar awakenedBar =
+            new TrackedBar(Pattern.compile("§fAwakening §7\\[§f(\\d+)/(\\d+)§7]"), TrackedBar.BarType.AWAKENED) {
+                @Override
+                public void onUpdateName(Matcher match) {
+                    try {
+                        current = Integer.parseInt(match.group(1));
+                        max = Integer.parseInt(match.group(2));
+                    } catch (NumberFormatException e) {
+                        WynntilsMod.error(String.format(
+                                "Failed to parse current and max for awakened bar %s (%s out of %s)",
+                                type, match.group(1), match.group(2)));
+                    }
+                }
+            };
+
+    public static final TrackedBar focusBar =
+            new TrackedBar(Pattern.compile("§eFocus §6\\[§e(\\d+)/(\\d+)§6]"), TrackedBar.BarType.FOCUS) {
+                @Override
+                public void onUpdateName(Matcher match) {
+                    try {
+                        current = Integer.parseInt(match.group(1));
+                        max = Integer.parseInt(match.group(2));
+                    } catch (NumberFormatException e) {
+                        WynntilsMod.error(String.format(
+                                "Failed to parse current and max for focus bar %s (%s out of %s)",
+                                type, match.group(1), match.group(2)));
+                    }
+                }
+            };
+
+    public static final TrackedBar corruptedBar =
+            new TrackedBar(Pattern.compile("§cCorrupted §4\\[§c(\\d+)%§4]"), TrackedBar.BarType.CORRUPTED) {
+                @Override
+                public void onUpdateName(Matcher match) {
+                    try {
+                        current = Integer.parseInt(match.group(1));
+                        max = 100;
+                    } catch (NumberFormatException e) {
+                        WynntilsMod.error(String.format(
+                                "Failed to parse current and max for corrupted bar %s (%s out of %s)",
+                                type, match.group(1), match.group(2)));
+                    }
+                }
+            };
+
+    private static final HashMap<UUID, TrackedBar> trackedBarsMap = new HashMap<>();
+
+    // FixPacketBugsFeature gets in the way if receiveCanceled is not set
```

Comment:
Is it still needed? If Wynncraft has remade boss bars they might have fixed the bug that sent broken packets as well.

Reply:
I think this would be better off as as separate pr, as it is getting out of scope.